packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,306 @@
+          value: name,
+        }))
+        .concat({
+          name: "I'm done",
+          value: abortKey,
+        }),
+    };
+    const { functionName } = await inquirer.prompt(functionNameQuestion);
+    if (functionName === abortKey) return;
+    await envVarSelectKey(functionName);
+  };
+
+  const envVarSelectKey = async (functionName: string) => {
+    const envVars = getStoredEnvironmentVariables(context, functionName, currentEnvName);
+    const abortKey = uuid.v4();
+    const keyNameQuestion: inquirer.ListQuestion = {
+      type: 'list',
+      name: 'keyName',
+      message: 'Which function’s environment variables do you want to edit?',
+      choices: Object.keys(envVars)
+        .map(name => ({
+          name,
+          value: name,
+        }))
+        .concat({
+          name: "I'm done",
+          value: abortKey,
+        }),
+    };
+    const { keyName } = await inquirer.prompt(keyNameQuestion);
+    if (keyName === abortKey) {
+      await envVarSelectFunction();
+      return;
+    }
+    await envVarUpdateValue(functionName, keyName);
+  };
+
+  const envVarUpdateValue = async (functionName: string, keyName: string) => {
+    const envVars = getStoredEnvironmentVariables(context, functionName, currentEnvName);
+    const newValueQuestion: inquirer.InputQuestion = {
+      type: 'input',
+      name: 'newValue',
+      message: 'Enter the environment variable value:',
+      default: envVars[keyName],
+      validate: input => {
+        if (input.length >= 2048) {
+          return 'The value must be 2048 characters or less';
+        }
+        return true;
+      },
+    };
+    const { newValue } = await inquirer.prompt(newValueQuestion);
+    functions[functionName][_.camelCase(keyName)] = newValue;
+    await envVarSelectKey(functionName);
+  };
+
+  await envVarQuestion();
+  Object.keys(functions).forEach(functionName => {
+    setStoredKeyValue(context, functionName, functions[functionName]);
+  });
+};
+
+export const askEnvironmentVariableToPush = async (context: $TSContext, currentEnvName: string, projectPath: string) => {
+  const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, {
+    throwIfNotExist: false,
+    default: {},
+  });
+  const functions = teamProviderInfo[currentEnvName]?.categories?.function || {};
+  const functionNames = Object.keys(functions);
+  if (functionNames.length === 0) {
+    return;
+  }
+  context.print.info('');
+  context.print.info('Some Lambda function environment variables are defined but missing values.');
+  context.print.info('');
+  for (const [_, functionName] of functionNames.entries()) {
+    const storedList = getStoredList(functionName);
+    const storedKeyValue = getStoredKeyValue(context, functionName);
+    for (const [_, item] of storedList.entries()) {
+      const envName = item.environmentVariableName;
+      const keyName = item.cloudFormationParameterName;
+      if (storedKeyValue.hasOwnProperty(keyName)) return;
```

Comment:
this should be a `continue` not a return

Reply:
Sounds good. I replaced return with continue :)